lib/kubeclient/common.rb

```diff
@@ -50,21 +36,23 @@ def initialize_client(
     def handle_exception
       yield
     rescue RestClient::Exception => e
-      begin
-        json_error_msg = JSON.parse(e.response || '') || {}
+      json_error_msg = begin
+        JSON.parse(e.response || '') || {}
       rescue JSON::ParserError
-        json_error_msg = {}
+        {}
       end
       err_message = json_error_msg['message'] || e.message
       raise KubeException.new(e.http_code, err_message, e.response)
     end
 
     def handle_uri(uri, path)
-      fail ArgumentError, 'Missing uri' if uri.nil?
-      @api_endpoint = (uri.is_a? URI) ? uri : URI.parse(uri)
-      @api_endpoint.path = path if @api_endpoint.path.empty?
-      @api_endpoint.path = @api_endpoint.path.chop \
-                         if @api_endpoint.path.end_with? '/'
+      fail ArgumentError, 'Missing uri' unless uri
+      @api_endpoint = (uri.is_a?(URI) ? uri : URI.parse(uri))
+      if @api_endpoint.path.empty?
+        @api_endpoint.path = path
+      elsif @api_endpoint.path.end_with? '/'
```

Comment:
yeah true ... somehow got that mixed up and thought it would only call chop on the empty string